### PR TITLE
Add virtual destructors and make classes be `final`

### PR DESCRIPTION
**Pull request by onelxj**

### CODE REVIEW

File: eventuals/concurrent-ordered.h [hunk 1]
```diff
@@ -20,12 +20,14 @@ namespace eventuals {
...
 
-struct _ReorderAdaptor {
+struct _ReorderAdaptor final {
   template <typename K_, typename Value_>
-  struct Continuation : public TypeErasedStream {
+  struct Continuation final : public TypeErasedStream {
     Continuation(K_ k)
       : k_(std::move(k)) {}
 
+    Continuation(Continuation&& that) = default;
```
Comment (benh) [inline eventuals/concurrent-ordered.h:29]: Did a compiler ask for this?

File: eventuals/event-loop.h [hunk 2]
```diff
@@ -420,7 +420,7 @@ class EventLoop : public Scheduler {
 
   EventLoop();
   EventLoop(const EventLoop&) = delete;
-  virtual ~EventLoop();
+  ~EventLoop();
```
Comment (benh) [inline eventuals/event-loop.h:423]:
```suggestion
  ~EventLoop() override;
```

File: eventuals/range.h [hunk 3]
```diff
@@ -10,7 +10,7 @@ namespace eventuals {
 
 struct _Range {
```
Comment (benh) [inline eventuals/range.h:11]: This one too?

File: eventuals/scheduler.h [hunk 4]
```diff
@@ -53,7 +53,7 @@ class Scheduler {
 
     Context(Context&& that) = delete;
 
-    virtual ~Context() {}
+    virtual ~Context() = default;
```
Comment (benh) [inline eventuals/scheduler.h:56]: Let's actually make `Context` be `final`!

**Review by benh — CHANGES_REQUESTED** [review]
Please add `override` to destructors of derived classes, thanks!

File: eventuals/concurrent-ordered.h (hunk 1 above)
Comment (onelxj) [reply eventuals/concurrent-ordered.h:29]: Yep

File: eventuals/range.h (hunk 3 above)
Comment (onelxj) [reply eventuals/range.h:11]: Thanks!

File: eventuals/scheduler.h (hunk 4 above)
Comment (onelxj) [reply eventuals/scheduler.h:56]: Let's chat about that synchronously.

Comment (benh) [inline eventuals/scheduler.h:56]: Doesn't need to be `virtual` because it's `final` and doesn't `override` any base class destructor!

**Review by benh — APPROVED** [review]

**Pull request merged**